enhancements/cnf-tests-split.md

```diff
@@ -0,0 +1,204 @@
+Running the `e2esuite` test suite, with an empty `CNF_COMPONENTS` and `FEATURES=metallb`, will result in 
+running ginkgo's test in all of the components: sriov, metallb, nto-performance, nto-latency, ptp and integration, adding the flag `-ginkgo.focus=metallb`.
+So in other words, the functionallity of the `FEATURES` var stays the same.
+
+When running the `e2esuite` test suite with `CNF_COMPONENTS=metallb` it will invoke only the metallb component
+of the e2esuite, i.e.: `cnf-tests/submodules/metallb-operator/test/e2e/functional/tests`
+and for the validation suite, it will run: `cnf-tests/submodules/metallb-operator/test/e2e/validation/tests`
+
+The mapping of suite + component to the external test suite path is done by:
+```bash
+declare -A TESTS_PATHS=\
+(["configsuite nto"]="cnf-tests/submodules/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/0_config"\
+ ["validationsuite cluster"]="cnf-tests/testsuites/validationsuite"
+ ["validationsuite metallb"]="cnf-tests/submodules/metallb-operator/test/e2e/validation"
+ ["cnftests integration"]="cnf-tests/testsuites/e2esuite"\
+ ["cnftests metallb"]="cnf-tests/submodules/metallb-operator/test/e2e/functional"\
+ ["cnftests sriov"]="cnf-tests/submodules/sriov-network-operator/test/conformance"\
+ ["cnftests nto-performance"]="cnf-tests/submodules/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance"\
+ ["cnftests nto-latency"]="cnf-tests/submodules/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/4_latency"\
+ ["cnftests ptp"]="cnf-tests/submodules/ptp-operator/test/conformance/serial")
+```
+
+To invoke the cnf-tests we'll use the same `Makefile` targets as before. e.g.: `functests-on-ci`, `feature-deploy-on-ci`, etc.
+Those rules will have a prerequisite to run the new `init-git-submodules` target which inits and updates the git submodules.
+The `init-git-submodules` will include the following env vars that control the commit hash of the components:
+```bash
+export METALLB_OPERATOR_TARGET_COMMIT?= main
+export SRIOV_NETWORK_OPERATOR_TARGET_COMMIT?= master
+export PTP_OPERATOR_TARGET_COMMIT?= master
+export CLUSTER_NODE_TUNING_OPERATOR_TARGET_COMMIT?= master
+```
+
+The underline script that in charge of invoking the tests will be `run-functests.sh` instead of
+`cnf-tests/entrypoint/test-run.sh`
+
+### Reports
+
+In order to maintain the same reporting capabilities we need to:
+ 1. Add the k8s reporter to each repository that doesn't has it already.
+ 2. Use ginkgo's built-in junit generator, so that we don't have to add the `-junit` flag to each repo.
+ 3. Create a local junit-merger program to aggregate all of the junit reports under the same test suite.
+ 4. Change the reports output directory. Having `TESTS_REPORTS_PATH` as the root path,
+   we will have to include "/raw" subpath, where we create all the junit reports, 
+   and a "/junit" subpath where we hold the merged junit reports.
+
+### Rewrite the cnf-tests container
```

Comment:
Just a clarification, we don't talk here about `quay.io/openshift-kni/cns-tests:..` container, that is used in performance tests? Or do we?

Reply:
Yes, it's that.